big_tests/tests/amp_big_SUITE.erl

```diff
@@ -647,6 +668,23 @@ last_rule_applies_test(Config) ->
 
 %% Internal
 
+wait_until_no_session(FreshConfig, User) ->
+    U = escalus_users:get_username(FreshConfig, User),
+    S = escalus_users:get_server(FreshConfig, User),
+    wait_until_no_session(U, S, 5000).
+
+wait_until_no_session(U, S, Timeout) ->
+    case length(rpc(mim(), ejabberd_sm, get_user_resources, [U, S])) of
+        0 ->
+            ok;
+        _ when Timeout > 0 ->
+            ct:log("Waiting ~p ms for MongooseIM to close the sessions for ~s@~s", [Timeout, U, S]),
+            ct:sleep(1000),
+            wait_until_no_session(U, S, Timeout - 1000);
+        _ ->
+            ct:fail("MongooseIM failed to close the session for ~s@~s")
+    end.
+
```

Comment:
There is `mongoose_helper:wait_until` function which helps to write such kind of waiting tests. Do you think it could be used here?

Reply:
This code is indeed ported from a very old commit, I will update this.